extension/src/testConfigManager.ts

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+import * as fs from 'fs';
+import * as path from 'path';
+import { window, workspace, ExtensionContext, TextEditor } from 'vscode';
+
+import { ProjectInfo, ProjectManager } from './projectManager';
+import * as Configs from './Constants/configs';
+import { TestConfig } from './Models/testConfig';
+import * as Logger from './Utils/Logger/logger';
+
+export class TestConfigManager {
+    private readonly _configPath: string;
+    constructor(context: ExtensionContext, private _projectManager: ProjectManager) {
+        this._configPath = context.asAbsolutePath(Configs.TEST_LAUNCH_CONFIG_NAME);
+    }
+
+    public get configPath(): string {
+        return this._configPath;
+    }
+
+    public loadConfig(): Promise<TestConfig> {
+        return new Promise((resolve, reject) => {
+            fs.access(this._configPath, (err) => {
+                if (err) {
+                    const config: TestConfig = this.createDefaultTestConfig();
+                    const content: string = JSON.stringify(config, null, 4);
+                    fs.writeFile(this._configPath, content, 'utf-8', (writeErr) => {
+                        if (writeErr) {
+                            Logger.error('Failed to create default test config!', {
+                                error: writeErr,
+                            });
+                            reject(writeErr);

```

Comment:
Should return directly; otherwise will be call resolve after reject

Reply:
For promise, the resolve call after reject would be ignored. But I would add return to avoid confusion